btc/src/main/kotlin/handler/btc/NewBtcClientRegistrationHandler.kt

```diff
@@ -0,0 +1,49 @@
+package handler.btc
+
+import iroha.protocol.Commands
+import mu.KLogging
+import org.bitcoinj.core.Address
+import org.bitcoinj.wallet.Wallet
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.stereotype.Component
+import provider.btc.network.BtcNetworkConfigProvider
+import sidechain.iroha.CLIENT_DOMAIN
+
+/**
+ * Class that is used to handle client registration commands
+ */
+@Component
+class NewBtcClientRegistrationHandler(
+    @Autowired private val btcNetworkConfigProvider: BtcNetworkConfigProvider
+) {
+    /**
+     * Handles newly registered Bitcoin addresses and adds addresses to current wallet object
+     */
+    fun handleNewClientCommand(command: Commands.Command, wallet: Wallet) {
+        if (!isNewClientWasRegistered(command)) {
+            return
+        }
+        val address = Address.fromBase58(
+            btcNetworkConfigProvider.getConfig(),
```

Comment:
Are these configs changeable in runtime? If not, can we just `getConfig()` during initialization and reuse only that value?

Reply:
The value returned by `getConfig()` is a singleton.